health/health.go

```diff
@@ -85,15 +88,21 @@ func check(
 }
 
 // health checks the given health service
-func health(healthService string) bool {
-	resp, _ := http.Get(healthService)
-	bytes, _ := ioutil.ReadAll(resp.Body)
-
-	resp.Body.Close()
+func health(healthService string, httpTimeout time.Duration) bool {
+	client := http.Client{
+		Timeout: httpTimeout,
+	}
 
-	if resp == nil {
+	resp, err := client.Get(healthService)
+	// Check if response timeouts or returns an HTTP error
+	if err != nil {
 		return false
 	}
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false
+	}
+	resp.Body.Close()
```

Comment:
If there is an error in reading the response body, then the body will not be closed.  Probably we should just `defer resp.Body.Close()` before we try to read the body.